src/messaging/ExchangeMessageDispatch.cpp

```diff
@@ -80,7 +80,19 @@ CHIP_ERROR ExchangeMessageDispatch::SendMessage(SessionHandle session, uint16_t
         std::unique_ptr<ReliableMessageMgr::RetransTableEntry, decltype(deleter)> entryOwner(entry, deleter);
 
         ReturnErrorOnFailure(PrepareMessage(session, payloadHeader, std::move(message), entryOwner->retainedBuf));
-        ReturnErrorOnFailure(SendPreparedMessage(session, entryOwner->retainedBuf));
+        CHIP_ERROR err = SendPreparedMessage(session, entryOwner->retainedBuf);
+        if (err == System::MapErrorPOSIX(errno))
```

Comment:
Please make the condition specifically on some known errno numbers. As I read this, it says:

"if the error sending is an O/S serror,  then assume success". This seems incorrect for things like EBADF, EINVAL, ENOTSOCK and likely others.

Reply:
Er, this is just wrong, clearly.  It should be `System::MapErrorPOSIX(ENOBUFS)`.  Good catch!